tests/test_initialization.py

```diff
@@ -1846,6 +1847,98 @@ def test_road_with_conv2d_layer(self):
+                X = self.lin1(X)
+                return X
+
+        return MLP(bias=bias).to(self.torch_device).eval()
+
+    @pytest.fixture
+    def data(self):
+        torch.manual_seed(0)
+        return torch.randn(4, 10, device=self.torch_device)
+
+    def test_delora_injection_keeps_output_default(self, data):
+        # With use_residual_init=True (default), initial forward should match base model
+        torch.manual_seed(0)
+        base = self.get_model()
+        y_base = base(data)
+
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+        y_peft = model(data)
+
+        assert torch.allclose(y_base, y_peft, atol=1e-6, rtol=1e-6)
+
+    def test_delora_param_shapes(self):
+        base = self.get_model()
+        in_f, out_f = base.lin0.in_features, base.lin0.out_features
+        r = 4
+        cfg = DeloraConfig(target_modules=["lin0"], r=r, lambda_=2)
+        model = get_peft_model(base, cfg)
+
+        layer = model.lin0  # DeloraLinear wrapper
+        assert hasattr(layer, "delora_A") and hasattr(layer, "delora_B") and hasattr(layer, "delora_lambda")
+        A = layer.delora_A["default"]
+        B = layer.delora_B["default"]
+        lambda_ = layer.delora_lambda["default"]
+        assert tuple(A.shape) == (r, in_f)
+        assert tuple(B.shape) == (out_f, r)
+        assert tuple(lambda_.shape) == (1,)
+
+    def test_disable_enable_no_change(self, data):
+        base = self.get_model()
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+
+        y0 = model(data)
+        model.disable_adapter_layers()
+        y1 = model(data)
+        model.enable_adapter_layers()
+        y2 = model(data)
+
+        assert torch.allclose(y0, y1, atol=1e-6, rtol=1e-6)
+        assert torch.allclose(y0, y2, atol=1e-6, rtol=1e-6)
+
+    def test_merge_and_unload_same_outputs(self, data):
+        base = self.get_model()
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8)
+        model = get_peft_model(base, cfg)
+        y_before = model(data)
+
+        merged = model.merge_and_unload()
+        merged.eval()
+        y_after = merged(data)
+        assert torch.allclose(y_before, y_after, atol=1e-6, rtol=1e-6)
+
+    def test_invalid_rank_raises(self):
+        base = self.get_model()
+        with pytest.raises(ValueError):
+            get_peft_model(base, DeloraConfig(target_modules=["lin0"], r=0))
+
+    def test_no_use_residual_init_keeps_output(self, data):
+        # With use_residual_init=False, the implementation compensates by subtracting initial delta in get_delta
+        base = self.get_model()
+        y_base = base(data)
+        cfg = DeloraConfig(target_modules=["lin0"], r=8, lambda_=8, use_residual_init=False)
+        model = get_peft_model(base, cfg)
+        y_peft = model(data)
+        assert torch.allclose(y_base, y_peft, atol=1e-6, rtol=1e-6)
+
```

Comment:
Let's add a test adding a second adapter (via `model.add_adapter`) that has `use_residual_init=True` raises the "supports only 1 adapter with residual init" error.